### ISSUE TITLE
perldoc dies if a stray Makefile.PL is in cwd

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#2985]
Migrated from [rt.perl.org#4907](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=4907) (status was 'resolved')

Searchable as RT4907$

**Comment by p5pRT** [comment]
### From bgriffin@cddb.com

#### Created by bgriffin@cddb.com



$ mkdir /tmp/foo
$ cd /tmp/foo
$ touch Makefile\.PL
$ perldoc perl
Cannot find blib even in /tmp/foo/\.\./\.\./\.\./\.\./\.\.
BEGIN failed\-\-compilation aborted at \(eval 2\) line 1\.
&nbsp;    \.\.\.propagated at /usr/bin/perldoc line 165\.
$

<details><summary>Perl Info</summary>

```

Flags:
    category=utilities
    severity=medium

Site configuration information for perl v5.6.0:

Configured by bgriffin at Fri Jul 28 14:55:44 PDT 2000.

Summary of my perl5 (revision 5.0 version 6 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.2.5-15, archname=i686-linux-thread-multi
    uname='linux bgriffin-unx 2.2.5-15 #1 mon apr 19 23:00:46 edt 1999 i686 unknown '
    config_args=''
    hint=previous, useposix=true, d_sigaction=define
    usethreads=define use5005threads=define useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define 
    use64bitint=undef use64bitall=undef uselongdouble=define usesocks=undef
  Compiler:
    cc='cc', optimize='-O2', gccversion=egcs-2.91.66 19990314/Linux (egcs-1.1.2 release)
    cppflags='-D_REENTRANT -fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    ccflags ='-D_REENTRANT -fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    stdchar='char', d_stdstdio=define, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='long double', nvsize=12, Off_t='off_t', lseeksize=8
    alignbytes=4, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lnsl -lndbm -lgdbm -ldb -ldl -lm -lpthread -lc -lposix -lcrypt
    libc=/lib/libc-2.1.1.so, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.6.0:
    /home/bgriffin/perl5lib
    /usr/lib/perl5/5.6.0/i686-linux-thread-multi
    /usr/lib/perl5/5.6.0
    /usr/lib/perl5/site_perl/5.6.0/i686-linux-thread-multi
    /usr/lib/perl5/site_perl/5.6.0
    /usr/lib/perl5/site_perl/5.005
    /usr/lib/perl5/site_perl
    .


Environment for perl v5.6.0:
    HOME=/home/bgriffin
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/bgriffin/bin:/bin:/usr/bin:/usr/local/bin:/usr/X11R6/bin:/sbin:/usr/bin:/usr/sbin
    PERL5LIB=/home/bgriffin/perl5lib
    PERL_BADLANG (unset)
    SHELL=/bin/ksh


```
</details>

**Comment by p5pRT** [comment]
### From @simoncozens

On Wed\, Dec 13\, 2000 at 05&#8203;:16&#8203;:44PM \-0800\, Benjamin Elijah Griffin wrote&#8203;:
> $ perldoc perl
> Cannot find blib even in /tmp/foo/\.\./\.\./\.\./\.\./\.\.
> BEGIN failed\-\-compilation aborted at \(eval 2\) line 1\.
>     \.\.\.propagated at /usr/bin/perldoc line 165\.

This one's been fixed&#8203;:

&nbsp;    \# don't add if superuser
&nbsp;    if \($\< && $> && \-f "blib"\) \{   \# don't be looking too hard now\!
&nbsp;    eval q\{ use blib; 1 \};
&nbsp;    warn $@&#8203; if $@&#8203; && $opt\_v;
&nbsp;    \}
